core/src/main/java/org/elasticsearch/transport/TransportActionProxy.java

```diff
@@ -53,23 +53,25 @@ private TransportActionProxy() {} // no instance
         public void messageReceived(T request, TransportChannel channel) throws Exception {
             DiscoveryNode targetNode = request.targetNode;
             TransportRequest wrappedRequest = request.wrapped;
-            service.sendRequest(targetNode, action, wrappedRequest, new ProxyResponseHandler<>(channel, responseFactory));
+            service.sendRequest(targetNode, action, wrappedRequest, new ProxyResponseHandler<>(channel, wrappedRequest, responseFactory));
```

Comment:
do we really need to hold on to the request or can we invoke the function here? I really think we shouldn't hold on to the request?!